### PR TITLE
fix json in describe component

### PR DESCRIPTION
**Pull request by kadel**
**What type of PR is this?**

/kind bug

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/openshift/odo/issues/4474
Fixes https://github.com/openshift/odo/issues/4463

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

run `odo catalog describe component nodejs -o json` 
the output should be a valid json struct (array)

### CODE REVIEW

**Comment by kadel** [comment]
/retest
timeouts

**Comment by prietyc123** [comment]
@kadel Can we validate this scenario by adding tests with similar behaviour?

**Review by prietyc123 — CHANGES_REQUESTED** [review]
https://github.com/openshift/odo/pull/4506#issuecomment-797377611

**Comment by kadel** [comment]
> @kadel Can we validate this scenario by adding tests with similar behaviour?

done

File: pkg/odo/cli/catalog/describe/component.go [hunk 1]
```diff
@@ -103,19 +105,28 @@ func (o *DescribeComponentOptions) Validate() (err error) {
 	return nil
 }
 
+// DevfileComponentDescription represents the JSON output of Devfile compoent description
```
Comment (mik-dass) [inline pkg/odo/cli/catalog/describe/component.go:108]: `component`

Comment (kadel) [reply pkg/odo/cli/catalog/describe/component.go]: fixed

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**